Source/WebCore/testing/Internals.cpp

```diff
@@ -6827,4 +6827,19 @@ String Internals::modelInlinePreviewUUIDForModelElement(const HTMLModelElement&
 
 #endif
 
+void Internals::avoidIOSurfaceSizeCheckInWebProcess()
+{
+    auto* document = contextDocument();
+    if (!document)
+        return;
+    auto* page = document->page();
+    if (!page)
+        return;
+    page->settings().setMaximumAccelerated2dCanvasSize(UINT_MAX);
+    HTMLCanvasElement::setMaxCanvasAreaForTesting(UINT_MAX);
+    HTMLCanvasElement::setMaxPixelMemoryForTesting(UINT_MAX);
+    page->chrome().client().avoidIOSurfaceSizeCheckInWebProcess();
```

Comment:
It is weird that we set 'avoidIOSurfaceSizeCheckInWebProcess' in RemoteRenderingBackendProxy via ChromeClient but use it only once for the first RemoteImageBufferProxy we create. Can't we apply it to the canvas element and propagate it from the HTMLCanvasElement to its RemoteImageBufferProxy::create()?
`
void Internals::avoidIOSurfaceSizeCheckInWebProcess(HTMLCanvasElement& element)
`

Reply:
Fixed.